va/validation-authority_test.go

```diff
@@ -213,12 +218,19 @@ func TestSimpleHttpTLS(t *testing.T) {
 	finChall, err := va.validateSimpleHTTP(ident, chall)
 	test.AssertEquals(t, finChall.Status, core.StatusValid)
 	test.AssertNotError(t, err, chall.Path)
+	logs := log.GetAllMatching(`^\[AUDIT\] Attempting to validate SimpleHTTPS for `)
+	test.AssertEquals(t, len(logs), 1)
+	test.AssertEquals(t, logs[0].Priority, syslog.LOG_NOTICE)
 }
 
 func TestSimpleHttp(t *testing.T) {
 	va := NewValidationAuthorityImpl(true)
 	va.DNSResolver = &mocks.MockDNS{}
 
+	log := mocks.NewSyslogWriter()
+	defer log.Close()
+	defer mocks.SwitchLog(mocks.SwitchLog(log))
+
```

Comment:
The goal here is to restore the default singleton SyslogWriter after a test? I think, as you say, it's easier to assume that the state of the singleton SyslogWriter is undefined at the beginning of any test, and make sure to override it with UseMockLog for any test that relies on checking log output.